File: Array/Python/Uncommon Words from Two Sentences.py
# Problem link: https://leetcode.com/problems/uncommon-words-from-two-sentences/description/
# Time: O(M + N)
# Space: O(M + N)
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        store = Counter(s1.split() + s2.split())
        result = []

        for key, val in store.items():
            if val == 1:
                result.append(key)
        
        return result

File: Array/Java/Uncommon Words from Two Sentences.java
// Problem link: https://leetcode.com/problems/uncommon-words-from-two-sentences/description/
// Time: O(M + N)
// Space: O(M + N)
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        Map<String, Integer> count = new HashMap();

        for(String word: s1.split(" "))
            count.put(word, count.getOrDefault(word, 0) + 1);

        for(String word: s2.split(" "))
            count.put(word, count.getOrDefault(word, 0) + 1);

        List<String> result = new LinkedList();

        for(String word: count.keySet())
            if (count.get(word) == 1)
                result.add(word);

        return result.toArray(new String[result.size()]);
    }
}
